server/pages/index.js

```diff
@@ -300,6 +301,24 @@ function setUpLoggedInRoute( req, res, next ) {
 		'X-Frame-Options': 'SAMEORIGIN',
 	} );
 
+	const LANG_REVISION_FILE_URL = 'https://widgets.wp.com/languages/calypso/lang-revisions.json';
+	const langPromise = superagent
+		.get( LANG_REVISION_FILE_URL )
+		.then( response => {
+			const languageRevisions = response.body;
+
+			req.context.languageRevisions = languageRevisions;
+
+			return languageRevisions;
+		} )
+		.catch( error => {
+			console.error( 'Failed to fetch the language revision files.', error );
+
+			return error;
+		} );
```

Comment:
I know I suggested moving `langPromise` up here, but isn't invoking its `.then()` and `.catch()` clauses going to run it immediately? I.e. I think if we don't want to execute it immediately, we can only pass `superagent.get( LANG_REVISION_FILE_URL )` to `Promises.all()`, where the network `response` will be passed as the first arg to the `.then`?

Reply:
(I might be mistaken here, resolving the promise early might work just fine, if I'm reading `Promise.all()` docs corectly.)

Reply:
It's intentional to run it earlier. As you already noted, `Promise.all()` will act as the safe guard here whether the both promises been resolved earlier or not.